### PR TITLE
[Merged by Bors] - Update Getting Started page to recommend debug mode optimizations

### PR DESCRIPTION
**Pull request by Roryl-c**
It seems to me that one of the more common stumbling blocks for new Bevy users is that when they complete the Getting Started guide and start on a project involving 2d or 3d scenes, their debug build performance suffers immensely. Some examples: 
 - https://github.com/bevyengine/bevy/issues/346
 - https://github.com/bevyengine/bevy/discussions/1410
 - https://github.com/bevyengine/bevy/issues/3027
 - [Why does Bevy have trouble drawing my many SpriteComponents?](https://stackoverflow.com/questions/63590141/why-does-bevy-have-trouble-drawing-my-many-spritecomponents)

This PR adds a section above the "Enable Fast Compiles" section of the Setup guide entitled "Compile with Performance Optimizations" with content adapted from the [Bevy Cheat Book section](https://bevy-cheatbook.github.io/pitfalls/performance.html) addressing this common issue (content provided under [MIT-0](https://github.com/bevy-cheatbook/mit-0) for [contributions to this repo](https://bevy-cheatbook.github.io/introduction.html#license)).

### CODE REVIEW

File: content/learn/book/getting-started/setup/_index.md
Comment (alice-i-cecile) [inline content/learn/book/getting-started/setup/_index.md:137]: I really don't think this should be optional. This is effectively required for anything beyond the most basic projects.

As a result I think it should come earlier than fast compiles.

File: content/learn/book/getting-started/setup/_index.md
Comment (alice-i-cecile) [inline content/learn/book/getting-started/setup/_index.md:139]:
```suggestion
While it may not be an issue for simple projects, debug builds in Rust can be _very slow_ - especially when you start using Bevy to make real games.
```

File: content/learn/book/getting-started/setup/_index.md
Comment (alice-i-cecile) [inline content/learn/book/getting-started/setup/_index.md:137]:
```suggestion
### Compile with Performance Optimizations
```

File: content/learn/book/getting-started/setup/_index.md
Comment (IceSentry) [inline content/learn/book/getting-started/setup/_index.md]: @inodentry are you ok with those comments being copied here? If not @Roryl-c this should be either removed or changed to something else.

**Review by alice-i-cecile — COMMENTED** [review]
In favor of this! I have some nits. The changes to the rest of this page look correct, but they are a bit noisy.

**Review by IceSentry — APPROVED** [review]
I'm not sure what's going on with the diff since the fast compile section seems like it's the same to me.

There's the small potential copyright issue, but other than that looks good to me

**Comment by Roryl-c** [comment]
Sorry about those noisy changes, I didn't realize my formatter would run on the file and didn't notice the changes before opening the PR. Should be cleaned up now.

@IceSentry Before opening this I checked what the licensing situation was with the Bevy Cheat Book, and the wording in the [License section](https://bevy-cheatbook.github.io/introduction.html#license) seems to ensure that content from the Cheat Book can be used in the official Bevy Project without any issue. I probably should have explicitly asked permission anyway though, as it's a little potentially rude - my bad on that @inodentry.

**Comment by IceSentry** [comment]
Ah yes, I forgot about that exception from the cheatbook, but it's always good to ask anyway!

**Review by alice-i-cecile — APPROVED** [review]

**Comment by DJMcNab** [comment]
Unfortunately, I believe this change is breaking the license of the cheatbook, as until it is merged it cannot be understood to be contributed to the repository:

Consider an adversarial licensee of the cheatbook, who opens a PR containing the entire cheatbook to this repository. It's clear in the author's use of a more restrictive license in general, that they do not intend for that to allow the content's license to be laundered until it is actually merged into the mainline.
This means that we must treat it as if we don't have the MIT-0 license whilst this PR is in an unmerged state.

Therefore, until the change is merged you need to ensure that the conditions of the only license you have to the content is followed, i.e. the `CC-BY-NC-SA`. This has attribution requirements, and the sharealike requirement should be mentioned in the top level readme (of your forked branch)

When and if this pr gets merged, an instant follow-up pr would be needed to remove this mention. Unfortunately, there is no period in the normal PR process for this to happen in an atomic change to the mainline, since anytime you remove these from your fork, it means your fork is not following the license.

(Additionally, when removing these comments, you should probably use a new branch from the mainline, rather than your current branch. This is because the version which has been contributed has an MIT-0 license, but your fork was only reached through the CC-BY-NC-SA path. I'm not quite sure how necessary this is, because licenses are messy)

**Comment by inodentry** [comment]
@DJMcNab honestly i feel like your attitude towards my licensing with cheatbook borders on harassment at this point.

I have made it abundantly clear many times, that my intention is for bevy contributors to be able to freely use and copy stuff from cheatbook without any worries about licensing or asking for permission. Everyone else seems to largely understand that, and seems to have no problem with it.

If every time cheatbook comes up in the context of someone wanting to contribute to bevy, you respond with some alarmist pedantic reading of my license exception, and pointing out how technically one could come up with loopholes in it, you aren't helping anyone. You are just making life harder for me and for Bevy contributors.

I want to help Bevy and its documentation to grow. I want cheatbook to feed into it. Please stop pedantically nitpicking my license. (Unless you have an actual suggestion for how I could make it clearer or better, given everything I've said regarding what my actual intentions are.)

*No bevy contributor is ever going to get in any sort of legal trouble because of basing their contributions on cheatbook.*

The license is as is, simply because I don't want another unofficial book effort or 3rd party project, that someone makes for their own gain or profit, to just take stuff from mine and/or misattribute it. I've really tried my best to make it as clear as possible that I want to be as permissive as possible towards Bevy and its contributors.

**Review by inodentry — APPROVED** [review]
I like this. Having this advice in the official book should be very valuable!

I'd add one sentence, just to quickly point out that adding optimizations to debug mode is preferable to just using release mode for development, because you don't lose debug assertions and other debug-mode-only functionality + compile times can be better.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by DJMcNab** [comment]
I have been coming from the standpoint of following the 'letter' of the exception, rather than to study intention.
That is, I want to ensure that I follow the requirements placed upon me by your license choice, regardless of what your intention is.
I feel this is necessary because in certain jurisdictions, including my own, breaking copyright *might* be criminal, regardless of the wishes of the copyright owner.
Although this is a non-issue in reality, I do not wish to be in the habit of wilfully committing criminal offenses.
You are correct though, that inforrming other users of this is unnecessarily inflammatory, and I intend to cease doing so.

Either the exception must be discretionary before the PR is merged, or the original license is toothless. I'm choosing to believe that you would not choose the original license if you want it to be toothless, so I am treating the exception must be discretionary.

A concrete improvement to the license exception would be to give a cure period (for when the contribution is submitted for inclusion in the listed repositories but the contribution has not been approved) as seen for example in the GPLv3.
I.e., require that you give an explicit notice of violation for non-approved contributions - such as a 'contribution' attempting to steal the entire cheatbook, and an amount of time to fix the violation in.
This would allow contributions to bevy be confident that they are not violating the copyright, so long as they can action a violation notice within the cure period you set out.

**Comment by alice-i-cecile** [comment]
@Roryl-c, I'm ready to merge this, if that's okay with you :)

**Comment by Roryl-c** [comment]
@alice-i-cecile Absolutely!

**Pull request closed without merging**
